### PR TITLE
Fix crashes for empty distro and fixes some react issue in component

### PR DESCRIPTION
**Pull request by MilanPospisil**
React component returned list of elements instead of one element, as it should return only one.

Distro had unresolved exception when distro was not found in the system.

### CODE REVIEW

File: frontend/hub/administration/repositories/RepositoryPage/RepositoryDetails.tsx [hunk 1]
```diff
@@ -38,16 +38,19 @@ export function RepositoryDetails() {
   if (distroError || remoteError)
     return <HubError error={distroError || remoteError} handleRefresh={refresh} />;
   if (!distroData || !remoteData) return <LoadingPage breadcrumbs tabs />;
-  const distribution: { name: string; client_url: string } = distroData?.results[0];
+
+  const distribution: { name: string; client_url: string } | undefined = distroData?.results[0]
```
Comment (himdel) [inline frontend/hub/administration/repositories/RepositoryPage/RepositoryDetails.tsx:42]: `maybeUndefined ? maybeUndefined : undefined` .. this doesn't do anything

```suggestion
  const distribution: { name: string; client_url: string } | undefined = distroData?.results[0];
```

File: frontend/hub/administration/repositories/components/RepositoryLabels.tsx [hunk 2]
```diff
@@ -9,16 +9,20 @@ export function RepositoryLabels(props: { repository: Repository }) {
...
+            <Label readOnly key={label}>
+              {label}: {repository.pulp_labels[label]}
+            </Label>
+          ) : (
+            <Label readOnly key={label}>
+              {label}
+            </Label>
+          )
+        )}
+      </>
```
Comment (himdel) [inline frontend/hub/administration/repositories/components/RepositoryLabels.tsx:25]: this just wraps the list of components in a fragment

it.. doesn't change anything either? Unless it's a typing issue and something can't cope with `ReactElement[]` even tough `ReactNode` allows it?

**Comment by himdel** [comment]
You're mentioning this is fixing some crashes but ... what are they?

File: frontend/hub/administration/repositories/components/RepositoryLabels.tsx (hunk 2 above)
Comment (MilanPospisil) [reply frontend/hub/administration/repositories/components/RepositoryLabels.tsx:25]: Change needed to fix typescript issues.

**Comment by MilanPospisil** [comment]
> You're mentioning this is fixing some crashes but ... what are they?

      <PageDetail label={t('Distribution')}>{distribution.name}</PageDetail>
      <PageDetail label={t('Distribution')}>{distribution?.name}</PageDetail>
      <PageDetail label={t('Repository URL')}>
        <CopyCell text={distribution.client_url} />
        <CopyCell text={distribution?.client_url} />

**Comment by MilanPospisil** [comment]
Already resolved in https://github.com/ansible/ansible-ui/pull/1453.

**Pull request closed without merging**